daemon/api.go

```diff
@@ -62,36 +62,42 @@ var api = []*Command{
 
 var (
 	rootCmd = &Command{
-		Path: "/",
-		GET:  SyncResponse([]string{"/1.0"}).Self,
+		Path:    "/",
+		GuestOK: true,
+		GET:     SyncResponse([]string{"/1.0"}).Self,
 	}
 
 	v1Cmd = &Command{
-		Path: "/1.0",
-		GET:  v1Get,
+		Path:    "/1.0",
+		GuestOK: true,
+		GET:     v1Get,
 	}
 
 	metaIconCmd = &Command{
-		Path: "/1.0/icons/{icon}",
-		GET:  metaIconGet,
+		Path:   "/1.0/icons/{icon}",
+		UserOK: true,
+		GET:    metaIconGet,
 	}
 
 	appIconCmd = &Command{
-		Path: "/1.0/icons/{name}.{origin}/icon",
-		GET:  appIconGet,
+		Path:   "/1.0/icons/{name}.{origin}/icon",
+		UserOK: true,
```

Comment:
information leak; you don't want joe random knowing what apps you have
installed.

On 27 November 2015 at 11:48, Zygmunt Krynicki <notifications@github.com>
wrote:

> In daemon/api.go
> <https://github.com/ubuntu-core/snappy/pull/180#discussion_r46040201>:
>
> >  	}
> >
> >  	appIconCmd = &Command{
> > -		Path: "/1.0/icons/{name}.{origin}/icon",
> > -		GET:  appIconGet,
> > +		Path:   "/1.0/icons/{name}.{origin}/icon",
> > +		UserOK: true,
>
> Why is this not guest OK?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snappy/pull/180/files#r46040201>.
>